lib/facter/package_provider.rb

```diff
@@ -12,10 +12,20 @@
 
 Facter.add(:package_provider) do
   setcode do
-    if Gem::Version.new(Facter.value(:puppetversion)) >= Gem::Version.new('3.6')
-      Puppet::Type.type(:package).newpackage(:name => 'dummy', :allow_virtual => 'true')[:provider].to_s
+    # if puppet enterprise is install use pe_version instead of puppetversion
+    if Facter.value(:pe_version) != nil
```

Comment:
This could all be simplified to:
```ruby
puppetversion = Facter.value(:pe_version) || Facter.value(:puppetversion)
if Gem::Version.new(puppetversion) > Gem::Version.new('3.6')
  Puppet::Type.type(:package).newpackage(:name => 'dummy', :allow_virtual => 'true')[:provider].to_s
else
  Puppet::Type.type(:package).newpackage(:name => 'dummy')[:provider].to_s
end
```